src/zkevm_specs/tx.py

```diff
@@ -0,0 +1,364 @@
+from .encoding import U8, is_circuit_code
+from typing import NamedTuple, Tuple, List, Sequence, Set
+from .util import FQ, RLC, U160, U256, U64
+from enum import IntEnum, auto
+from eth_keys import KeyAPI  # type: ignore
+import rlp  # type: ignore
+from eth_utils import keccak
+
+
+class Tag(IntEnum):
+    """
+    Tag used as second key in the Tx Circuit Rows to "select" the transaction field target.
+    Can be encoded in 4 bits.
+    """
+
+    Nonce = 1
+    Gas = 2
+    GasPrice = 3
+    GasTipCap = 4
+    GasFeeCap = 5
+    CallerAddress = 6
+    CalleeAddress = 7
+    IsCreate = 8
+    Value = 9
+    CallDataLength = 10
+    TxSignHash = 11
+    CallData = 12
+    Pad = 13
+
+
+class Row:
+    """
+    Tx circuit row
+    """
+
+    tx_id: FQ
+    tag: FQ
+    index: FQ
+    value: FQ
+
+    def __init__(self, tx_id: FQ, tag: FQ, index: FQ, value: FQ):
+        self.tx_id = tx_id
+        self.tag = tag
+        self.index = index
+        self.value = value
+
+
+class KeccakTable:
+    # The columns are: (is_enabled, input_rlc, input_len, output_rlc)
+    table: Set[Tuple[FQ, FQ, FQ, FQ]]
```

Comment:
1. I don't think we need `input_len` in the table, the consumer should be able to safely lookup without the `input_len` column.
2. Will add the `is_enabled` to the spec

Reply:
@han0110 I see why `input_len` is needed now. Will add to the spec.

Reply:
I completely agree with `input_len` indeed, we need in inside the permutations too to know how much of the entire input have we accomulated so far.

As for `is_enabled`. I understand the direction @ed255 wants to go towards.
Wondering if this can affect the lookups. As basically, it would make sense to define the "disabled" lookup as `input_len = 0`. 

In that case, we cannot hash something with zero length. And so, this could represent the disabled lookup.
And would save us one column.

WDYT?

Reply:
There will be situations where we need to hash a zero length array.  For example, when sending a tx to an existing account that has no code, there will be a verification that the `account.codeHash = keccak([])`.
If we don't have the empty array as an entry in the keccak lookup table, we'll need to add special cases in all callers to the keccak table that can encounter an empty array to hardcode the case for the empty array.  So I see this as a tradeoff.

Reply:
But that would require to do RLC(empty) which should be equal to `r` no? So the RLC result would be `r`.

Maybe we could use this property and use `[0x00;32]` as RLC input our default non active?

Reply:
> EDIT: This would only work if we use the RLC formula: `RLC = r + r*input[1] + r**2 * input[2] +.... + r**N * input[N]` so that since there's not going to be any real usage of keccak which is RLC(input) = 0.

Notice that this RLC formula is different than the one currently used elsewhere for Words.  Related discussion about RLC formula: https://github.com/appliedzkp/zkevm-specs/issues/135